src/ViewMarc.js

```diff
@@ -10,32 +10,45 @@ import {
 
 class ViewMarc extends React.Component {
   static manifest = Object.freeze({
+    query: {},
     marcRecord: {
       type: 'okapi',
       path: 'source-storage/formattedRecords/:{id}?identifier=INSTANCE',
-    }
+      accumulate: true,
+      throwErrors: false,
+    },
   });
 
+  componentDidUpdate() {
+    const { onClose } = this.props;
+
+    if (!this.props.marcRecord) {
+      this.props.mutator.marcRecord.GET()
+        .then(data => { this.props.marcRecord = data; })
```

Comment:
This is still an issue. You should never mutate props on a React component. It subverts the entire React philosophy of props-go-down-and-events-go-up. If you want to store something for use locally here, use `state`. If you want the updated `marcRecord` to be available to the parent, use a callback.

Reply:
I don't need it to be visible to a parent.
It's just a fallback for the case when the user reloads the page with the view record layer open.
In this case MARC Record generated on the parent and given the layer through props is not available any more, so I need to check if it's present and if it's not then try to retrieve it in place.

Reply:
But I see your point and will try to rework it to state usage.

Reply:
Done.